overlord/snapstate/snapstate.go

```diff
@@ -452,7 +452,17 @@ func InstallPath(st *state.State, si *snap.SideInfo, path, channel string, flags
 		instFlags |= skipConfigure
 	}
 
+	snapf, err := snap.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	info, err := snap.ReadInfoFromSnapFile(snapf, si)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
<3
I remember we weren't doing this for a good reason, but I forget why, and I suspect snap.Open resolves that reason. So, thank you: this will (let us) make errors from `snap install foo.snap` and `snap try` closer match errors from `snap install foo`.

Reply:
we have  backend.OpenSnapFile  to do this kind of thing, also it merits a comment that it's ok to do this here because either we have a SideInfo or this is the --dangerous case!

Reply:
Thanks to both of you! I improved the comment and use backend.OpenSnapFile() now.